### PR TITLE
updated release notes

### PR DESCRIPTION
**Pull request by AndreasHeger**

### CODE REVIEW

**Comment by AndreasHeger** [comment]
Thanks, @jmarshall , if you want to have a quick look at the release notes, please feel free to do so.

Special thanks for your work on the osX build will go into the github release notes!

File: NEWS [hunk 1]
```diff
@@ -5,18 +5,22 @@ http://pysam.readthedocs.io/en/latest/release.html
...
+support. Pysam is tested with python versions 3.7 to 3.11.
+
+* [#1175] VariantHeader.new_record: set start/stop before alleles
+* [#1173] Add multiple build improvements in htscodecs on multi-arch macOS
+* [#1148] Ignore CIGAR-less reads in find_introns.
+* [#1172] Add new `samtools cram-size` and `samtools reset` commands
+* [#1168] Add a minimal pyproject.toml for PEP517.
+* [#1158] Fix type hints and add FastqProxy type hints.
+* [#1147] Py3.11 compatibility, get shared object suffix from EXT_SUFFIX.
+* [#1143] Add mypy symbols for samtools and bcftools.
```
Comment (jmarshall) [inline NEWS:23]: These will be largely covered by the build improvements bit you mentioned you'll add to the GitHub release message, but as there are issue numbers for them that people might look for, we might add brief notes to NEWS too:

```
[#1155] Fix pysam.index() when using recent `samtools index` options.
[#1151] Test suite py3.11 compatibility, work around failing test case.
[#1149] MacOS universal build compatibility.
[#1146] Fix build when CFLAGS/etc environment variables are set.
```

**Review by jmarshall — COMMENTED** [review]
Looks good, a couple of suggestions.

**Comment by jmarshall** [comment]
> Pysam is now compatible with 3.11. We have removed python 2.x support. Pysam is tested with python versions 3.7 to 3.11.

It'd be good to have some clarity about exactly which Python 3.x versions we aim to support.

Currently (removed by this PR) _NEWS_ suggests the minimum is 3.2. _pyproject.toml_ says 3.8 (I hope this won't cause any trouble for people using 3.6 or 3.7). Our CI tests check 3.7 through 3.11, and only dropped 3.6 because GitHub's Ubuntu images no longer contain it.

I suspect many current extant HPC deployments' system Python installation is still 3.6 — e.g. various CentOS or Ubuntu installations.

So ideally I think we would say we support Py 3.6 or later. (Or stick with the existing 3.2 or later, but there's a few things missing from those old versions that's a bit painful.)

File: NEWS (hunk 1 above)
Comment (AndreasHeger) [reply NEWS:23]: Thanks, added!

**Comment by AndreasHeger** [comment]
That makes sense. I will do a quick check which python versions still work.

**Comment by jmarshall** [comment]
I fixed one more bug (#1169) on **master**, so have added a NEWS entry for it.

**Comment by AndreasHeger** [comment]
Thanks. I tested under 3.6, but failed getting any older python version to install and did not have time to setup separate test environments for older pythons.

Setting minimum python version to 3.6 and saying tested on 3.6 - 3.11.

**Pull request merged**